src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -2,32 +2,124 @@
 
 import java.util.List;
 
-public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+class Node<T> {
+    T data;
+    Node<T> prev;
+    Node<T> next;
+
+    Node(T data) {
+        this.data = data;
+        this.prev = null;
+        this.next = null;
+    }
+}
+
+
+class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
-    public void add(T value) {
+    public int size() {
+        return size;
     }
 
     @Override
-    public void add(T value, int index) {
+    public boolean isEmpty() {
+        return size == 0;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+
+        if (isEmpty()) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(
+                    "Index outside the list");
```

Comment:
Instead of throwing the exception directly in the `add` method, use the `checkIndex` method to handle index validation.